scalafix-rules/src/main/scala/scalafix/internal/rule/RemoveUnused.scala

```diff
@@ -153,6 +153,15 @@ class RemoveUnused(config: RemoveUnusedConfig)
         case Term.Function(List(param @ Term.Param(_, name, _, _)), _)
             if isUnusedParam(name.pos) =>
           Patch.replaceTree(param, "_")
```

Comment:
the fact that the predicate is on the name and the rewrite on the full param seems to be very specific to a 1-arity function with an implicit modifier, since in all other cases we want to match on the full param and replace only the name

Reply:
@taisukeoe can you comment on this since that's the test case you added? I had a look at https://www.scala-lang.org/files/archive/spec/2.13/06-expressions.html#anonymous-functions to try to understand how params can be implicit (and whether it's possible in a >1-arity function), but I am still confused.

Reply:
Thank you very much for elaboration. 
While current predicate works only when a param without explicit type, regardless of implicit type, it clearly misses a param with explicit type, because of the following behaviour. 

When `def g(x: String => Unit): Unit = ???` is given,

|matches with unused `s` diagnostics pos?|`g{s => ???}`|`g{implicit s => ???}`|`g{s: String => ???}`|
|:--|:--|:--|:--|
|param.name.pos|〇|〇|×|
|param.pos|〇|×|〇|

>  how params can be implicit (and whether it's possible in a >1-arity function)

As you pointed, 6.2.3 Anonymous Functions says:

> A named parameter of an anonymous function may be optionally preceded by an implicit modifier. In that case the parameter is labeled implicit; however the parameter section itself does not count as an implicit parameter section. Hence, arguments to anonymous functions always have to be given explicitly.

Although It's not very clearly said, `A named parameter of an anonymous function` might mean that it is allowed only for 1-arity function, and it matches what we experimented.